### PR TITLE
Adding docs for always decompile blocks

### PR DESCRIPTION
**Pull request by riknoll**

### CODE REVIEW

File: docs/blocks/typescript-blocks.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# JavaScript Block
+
+JavaScript is a very powerful language; so powerful that some concepts in
+JavaScript can't be shown using blocks. When PXT encounters a piece of code
+that can't be converted into blocks, it instead creates a grey JavaScript block
+to preserve it. These blocks get converted right back into the original
+JavaScript when you switch back to the text editor.
```
Comment (pelikhan) [inline docs/blocks/typescript-blocks.md]: screenshot...

**Comment by riknoll** [comment]
This is the help for the blocks, you can get there from the context menu

File: docs/blocks/typescript-blocks.md (hunk 1 above)
Comment (riknoll) [inline docs/blocks/typescript-blocks.md]: How do I add a screenshot?

Comment (pelikhan) [inline docs/blocks/typescript-blocks.md]: Add an image under /static/.... then use the markdown syntax, eg.

```
![](/static/mb/lessons/banana-keyboard-0.png)
```

**Comment by pelikhan** [comment]
should be linked from https://pxt.microbit.org/blocks too

**Comment by riknoll** [comment]
Sure thing

File: docs/blocks/typescript-blocks.md (hunk 1 above)
Comment (riknoll) [inline docs/blocks/typescript-blocks.md]: Actually, I can just add a block example

**Pull request merged**